src/WebJobs.Script/Workers/Rpc/Configuration/WorkerConfigurationResolver.cs

```diff
@@ -0,0 +1,276 @@
+                foreach (var probingPath in probingPaths)
+                {
+                    if (!string.IsNullOrEmpty(probingPath) && Directory.Exists(probingPath))
+                    {
+                        foreach (var languageWorkerPath in Directory.EnumerateDirectories(probingPath))
+                        {
+                            string languageWorkerDir = Path.GetFileName(languageWorkerPath);
+
+                            // If probing paths are malformed and have duplicate directories of the same language worker (eg. due to different casing)
+                            if (outputDict.ContainsKey(languageWorkerDir))
+                            {
+                                continue;
+                            }
+
+                            // Only skip worker directories that don't match the current runtime or are not enabled via hosting config
+                            if ((_workersAvailableForResolutionViaHostingConfig is not null &&
+                                !_workersAvailableForResolutionViaHostingConfig.Contains(languageWorkerDir)) ||
+                                (!_environment.IsMultiLanguageRuntimeEnvironment() &&
+                                !_environment.IsPlaceholderModeEnabled() &&
+                                workerRuntime is not null &&
+                                !workerRuntime.Equals(languageWorkerDir, StringComparison.OrdinalIgnoreCase)))
+                            {
+                                continue;
+                            }
+
+                            IEnumerable<string> workerVersions = Directory.EnumerateDirectories(languageWorkerPath);
+                            var versions = GetWorkerVersionsDescending(workerVersions);
+
+                            outputDict = GetWorkerConfigsFromProbingPaths(versions, languageWorkerPath, languageWorkerDir, releaseChannel, outputDict);
+                        }
+                    }
+                }
+            }
+
+            if (!_environment.IsMultiLanguageRuntimeEnvironment() &&
+                workerRuntime is not null &&
+                outputDict.ContainsKey(workerRuntime))
+            {
+                return outputDict.Values.ToList();
+            }
+
+            _logger.LogDebug("Searching for worker configs in the fallback directory.");
+
+            // Search in fallback path if worker cannot be found in probing paths
+            outputDict = GetWorkerConfigsFromWithinHost(fallbackPath, workerRuntime, outputDict);
+
+            return outputDict.Values.ToList();
+        }
+
+        private Dictionary<string, string> GetWorkerConfigsFromProbingPaths(IEnumerable<Version> versions, string languageWorkerPath, string languageWorkerFolder, string releaseChannel, Dictionary<string, string> outputDict)
+        {
+            int found = 0;
+
+            // language worker version
+            foreach (Version versionFolder in versions)
```

Comment:
this loop is a little hard to follow for me. could it be refactored to be something like 

if (releasechannel is null) {use the first in the list }
else if(realeasechannel == "latest") { use the first in the list }
else if(releasechannel == "standard") {use the second one (or first if not there)}
else if(releasechannel == "extended") {use the third, then second, then first, if not there}

... could have a helper that does the loop like GetCompatibleVersionAtIndex(int x).. which loops through and gets the xth compatitible version?

Reply:
We just need "n" and "n-1" versions here. I updated the code here to make it more readable.